### PR TITLE
[ASCII-1622] Prevent deadlock for status and flare command 

### PR DESCRIPTION
**Pull request by misteriaud**
### What does this PR do?

This PR add timeout to IPC connection between core Agent and other Agents (security-agent, trace-agent) 

### Motivation

In the current Agent state, when the port of an non-core Agent is taken by a process that hang incoming connections on it, any `datadog-agent status` or `datadog-agent flare` will stay block.

### Describe how to test/QA your changes

1. build the Agent
2. listen on the default port of the security-agent with netcat command: `nc -l 5010`
3. run the core Agent with `./bin/agent/agent run -c ./bin/agent/dist/datadog.yaml`
4. wait until a connection arrived to netcat
5. try to run the command `./bin/agent/agent status -c ./bin/agent/dist/datadog.yaml`
    a. before this PR the command must block
    b. with this fix the command must work (with at most 30 seconds of delay)

### CODE REVIEW

File: pkg/config/fetcher/from_processes.go [hunk 1]
```diff
@@ -23,9 +24,15 @@ func SecurityAgentConfig(config config.Reader) (string, error) {
 		return "", err
 	}
 
+	port := config.GetInt("security_agent.cmd_port")
+	if port <= 0 {
+		return "", fmt.Errorf("invalid security_agent.cmd_port -- %d", port)
+	}
+
 	c := util.GetClient(false)
+	c.Timeout = time.Second * 30
```
Comment (pgimalac) [inline pkg/config/fetcher/from_processes.go:33]: Can you add a config option for this timeout ?

File: releasenotes/notes/fix-flare-genereation-deadlock-b40fca3c82469d40.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fixed an issue where the datadog-agent status and datadog-agent flare commands would hang if a non-core Agent's port was occupied by a process that caused incoming connections to hang.
```
Comment (pgimalac) [inline releasenotes/notes/fix-flare-genereation-deadlock-b40fca3c82469d40.yaml:11]: I think a simpler description would make it clearer, what do you think ? (we don't really need to give a detailed explanation of the bug, just an overview)
```suggestion
    Fixed an issue where the ``datadog-agent status`` and ``datadog-agent flare`` commands could hang while trying to reach other agent processes.
```

File: pkg/config/fetcher/from_processes.go (hunk 1 above)
Comment (misteriaud) [reply pkg/config/fetcher/from_processes.go:33]: I can, but I think it's a really narrow use case and don't think having a config option is very useful.

Comment (GustavoCaso) [inline pkg/config/fetcher/from_processes.go:33]: Should we modify the `GetClient` to allow to get options. Maybe using the option pattern and [variadic arguments](https://gobyexample.com/variadic-functions)  so we can configure the http client `Timeout` directly inside?

**Review by git-thuerk-done — APPROVED** [review]
good for docs

File: pkg/config/fetcher/from_processes.go (hunk 1 above)
Comment (misteriaud) [reply pkg/config/fetcher/from_processes.go]: At the end I added it ! 👍

Comment (misteriaud) [reply pkg/config/fetcher/from_processes.go]: I had the same question, but configuring `http.Client` is very straightforward so I'm not sure of the gain of this modification.

File: pkg/config/setup/config.go [hunk 3]
```diff
@@ -973,6 +973,9 @@ func agent(config pkgconfigmodel.Config) {
 	config.SetKnown("proxy.http")
 	config.SetKnown("proxy.https")
 	config.SetKnown("proxy.no_proxy")
+
+	// Define the duration after which subagent configuration fetch will timeout
+	config.BindEnvAndSetDefault("subagent_config_fetch_timeout", "30s")
```
Comment (pgimalac) [inline pkg/config/setup/config.go:978]: I just remembered that there is already a config option somewhat for this: `server_timeout`
https://github.com/DataDog/datadog-agent/blob/f6d0c587696f58748dc04a5d4fd7ec54ee235f23/pkg/config/config_template.yaml#L749-L753

It's a timeout in the server, but IMO it makes sense to use it when querying the server too

Comment (misteriaud) [reply pkg/config/setup/config.go]: Done in fbaf2367b85c949e436f2e113abc67c570f018d5.

**Review by pgimalac — APPROVED** [review]
LGTM !

**Pull request merged**